lib/stimulus_reflex/reflex.rb

```diff
@@ -152,4 +119,45 @@ def render_collection(resource, content = nil)
     content ||= render(resource)
     tag.div(content.html_safe, id: dom_id(resource).from(1))
   end
+
+  private
+
+  def build_request
+    begin
+      uri = URI.parse(url)
+      path = ActionDispatch::Journey::Router::Utils.normalize_path(uri.path)
+      query_hash = Rack::Utils.parse_nested_query(uri.query)
+      mock_env = Rack::MockRequest.env_for(uri.to_s)
+
+      mock_env.merge!(
+        "rack.request.query_hash" => query_hash,
+        "rack.request.query_string" => uri.query,
+        "ORIGINAL_SCRIPT_NAME" => "",
+        "ORIGINAL_FULLPATH" => path,
+        Rack::SCRIPT_NAME => "",
+        Rack::PATH_INFO => path,
+        Rack::REQUEST_PATH => path,
+        Rack::QUERY_STRING => uri.query
+      )
+
+      env = connection.env.merge(mock_env)
+
+      middleware = StimulusReflex.config.middleware
+
+      if middleware.any?
+        stack = middleware.build(Rails.application.routes)
+        stack.call(env)
+      end
+
+      req = ActionDispatch::Request.new(env)
+
+      req.headers.merge!(headers)
```

Comment:
Moved back here as per this comment : https://github.com/stimulusreflex/stimulus_reflex/pull/538#issuecomment-902228070